### ISSUE TITLE
git-annex sync does not sync submodule state change commit

### ISSUE THREAD

**Issue by mih** [datalad/datalad#4292]
There seems to be no way to sync a submodule state change commit made in a adjusted branch into/with a corresponding branch. Here is the demo:

```sh
# main repository
mkdir subsync
cd subsync
git init
git annex init
# add some content to be able to switch into adjusted mode
touch 123; git add 123; git commit -m on
git annex adjust --unlock
# some more content
touch 234; git add 234; git commit -m two
# this is the desired type of sync call I need, it works fine (`git diff master` is empty afterwards)
git annex sync --no-commit --no-push --no-pull --no-content

# now make a submodule
mkdir sub; cd sub; git init; touch 123; git add 123; git commit -m one; cd ..
git submodule add ./sub; git commit -m three

# this type of sync is also fine (.gitmodules change, plus subproject state)
git annex sync --no-commit --no-push --no-pull --no-content

# now alter the submodule, and commit in the parent repo
cd sub; touch 234; git add 234; git commit -m two; cd ..
git commit -m four sub

# there is no way to sync this commit (plain subproject state change)
git annex sync --no-commit --no-push --no-pull --no-content
git annex sync

# still shows a diff
git diff master
```

@joeyh can you please confirm that this should work? Thx!

This functionality is required for #4275

**Comment by kyleam** [comment]
Filed [upstream](https://git-annex.branchable.com/bugs/Sync_of_adjusted_branch_does_not_propagate_changed_submodule_commit/).

**Comment by kyleam** [comment]
Resolved by git-annex's 376e69ec6 (adjust: Propagate submodule changes back to original branch, 2020-03-17).

**Comment by yarikoptic** [comment]
fwiw, 8.20200309+git133-gc29943f40 is being uploaded to debian-devel of neurodebian
